migrate/live-migration/live-migration-from-postgres.md

```diff
@@ -0,0 +1,252 @@
+<!-- TODO:  mention the housekeeping script that clears the buffered file s -->
+
+Before the stream of changes can be applied, the schema and data that existed
+prior to the creation of the replication slot must be migrated ([step 3][step-3]). The
+point that marks the beginning of the replication and buffering of changes is
+given by the exported snapshot. The larger the database, the more time it takes
+to perform the initial migration, and the longer the buffered files need to be
+stored.
+
+## 3. Migrate roles and schema from source to target
+
+### 3a. Dump the database roles from the source database
+
+<DumpDatabaseRoles />
+
+### 3b. Dump the database schema from the source database
+
+```sh
+pg_dump -d "$SOURCE" \
+  --format=plain \
+  --quote-all-identifiers \
+  --no-tablespaces \
+  --no-owner \
+  --no-privileges \
+  --schema-only \
+  --file=dump.sql \
+  --snapshot=$(cat /tmp/pgcopydb/snapshot)
+```
+
+- `--schema-only` is used to dump only the object definitions (schema), not
+  data.
+
+- `--snapshot` is use to specified the synchronized [snapshot][snapshot] when
+  making a dump of the database.
+
+<ExplainPgDumpFlags />
+
+### 3c. Load the roles and schema into the target database
+
+```sh
+psql -X -d "$TARGET" \
+  -v ON_ERROR_STOP=1 \
+  --echo-errors \
+  -f roles.sql \
+  -f dump.sql
+```
+
+## 4. Convert hypertables and enable Timescale features
+
+This is the ideal point to convert regular tables into hypertables. This step
+is optional. It can even be performed after all the data has been migrated,
+since hypertables can be created from tables with existing data.
+
+The advantage of converting before inserting the data is that the data is
+partitioned as it's written into the database. If done afterwards, the data
+has to first be written, then moved to the corresponding partition when the
+table is converted into a hypertable.
+
+A more detailed explanation can be found in the [hypertable
+documentation][hypertable-docs].
+
+In simple terms, you'll want to convert the tables that contain time series
+data. For each table that's going to be converted to a hypertable in the target
+database, run the following command:
+
+```sh
+psql -X -d "$TARGET" \
+  -v ON_ERROR_STOP=1 \
+  -c "SELECT create_hypertable('<table name>', '<time column name>')"
+```
+
+Once the table is converted, you can follow the guides to enable more Timescale
+features like [retention][retention-docs], [compression][compression-docs], and
+[data tiering][data-tiering-docs].
+
+## 5. Migrate initial data from source to target
+
+Use `pgcopydb` and the snapshot ID obtained in the second step to copy the
+data. The [snapshot][snapshot] ID is exported to `${TMPDIR}/pgcopydb/snapshot`.
+If `$TMPDIR` is not set, it's exported to `/tmp/pgcopydb/snapshot`.
+
+```sh
+pgcopydb copy table-data \
+  --source "$SOURCE" \
+  --target "$TARGET" \
+  --snapshot $(cat /tmp/pgcopydb/snapshot) \
+  --split-tables-larger-than <same-table concurrency size threshold>  \
```

Comment:
How should this be configured?

Reply:
@arajkumar can you check up on this?

Reply:
--split-tables-larger-than

Allow [Same-table Concurrency](https://pgcopydb.readthedocs.io/en/latest/design.html#same-table-concurrency) when processing the source database. This environment variable value is expected to be a byte size, and bytes units B, kB, MB, GB, TB, PB, and EB are known.


Reference: https://pgcopydb.readthedocs.io/en/latest/ref/pgcopydb_clone.html

Reply:
I think we should set this to a default like 10GB and add a note about it. WDYT?

Reply:
And table-jobs to 8?

Reply:
Yup, that sounds good